File: api_agent.py
from main import AgentRemote
import asyncio

remote_agent = AgentRemote("test", ["XYZ"], 10000)

async def main():
    last_trade = await remote_agent.get_latest_trade('XYZ')
    best_bid = await remote_agent.get_best_bid('XYZ')
    best_ask = await remote_agent.get_best_ask('XYZ')
    midprice = await remote_agent.get_midprice('XYZ')
    order_book = await remote_agent.get_order_book('XYZ')
    quotes = await remote_agent.get_quotes('XYZ')
    trades = await remote_agent.get_trades('XYZ')

    # market_buy = remote_agent.market_buy('XYZ', 1)
    # market_sell = remote_agent.market_sell('XYZ', 1)
    # limit_buy = remote_agent.limit_buy('XYZ', 1, 1)
    # limit_sell = remote_agent.limit_sell('XYZ', 1, 1)

    # get_price_bars = remote_agent.get_price_bars('XYZ', '1D', 20)
    print("last_trade", last_trade)
    print("best_bid", best_bid)
    print("best_ask", best_ask)
    print("midprice", midprice)
    print("order_book", order_book)
    print("quotes", quotes)
    print("trades", trades)

    # print("market_buy", market_buy)
    # print("market_sell", market_sell)
    # print("limit_buy", limit_buy)
    # print("limit_sell", limit_sell)
    
    # print("cancel_order", cancel_order)
    # print("cancel_all_orders", cancel_all_orders)
    
    # print("get_price_bars", get_price_bars)

# loop = asyncio.get_event_loop()
asyncio.run(main())

File: api_agent.js
const http = require('http')

const base = 'http://localhost:5000/api/v1'

count = 0

let functions = [
    get_latest_trade,
    get_best_bid,
    get_best_ask,
    get_midprice,
    get_order_book,
    get_quotes,
    get_trades
]

function hammer() {
    const start = Date.now()
    let runs = 100
    let run = 0
    let runners = []
    while (run < runs) {
        if(run >= runs) break
        run += 1
        runners.push(new Promise(resolve => setTimeout(() => {
            functions[Math.floor(Math.random() * functions.length)](resolve)
        }, 100) ) )
    }

    Promise.all(runners).then(() => {
        const end = Date.now()
        console.log(count, end - start)
    })
}

// make get request to api
function make_request(url, resolve) {
    http.get(url, response => {
        let body = ''
        response.on('data', data => {
            body += data.toString()
        })

        response.on('end', () => {
            let response = JSON.parse(body)
            resolve(response)
            count += 1
        })
    }
    )
}


function get_latest_trade(resolve) {
    const url =  `${base}/get_latest_trade?ticker=XYZ`
    make_request(url, resolve)
}

function get_best_bid(resolve) {
    const url =  `${base}/get_best_bid?ticker=XYZ`
    make_request(url, resolve)
}

function get_best_ask(resolve) {
    const url =  `${base}/get_best_ask?ticker=XYZ`
    make_request(url, resolve)
}

function get_midprice(resolve) {
    const url =  `${base}/get_midprice?ticker=XYZ`
    make_request(url, resolve)
}

function get_order_book(resolve) {
    const url =  `${base}/get_order_book?ticker=XYZ`
    make_request(url, resolve)
}

function get_quotes(resolve) {
    const url =  `${base}/get_quotes?ticker=XYZ`
    make_request(url, resolve)
}

function get_trades(resolve) {
    const url =  `${base}/get_trades?ticker=XYZ`
    make_request(url, resolve)
}


hammer()